src/zenml/integrations/great_expectations/materializers/ge_materializer.py

```diff
@@ -0,0 +1,119 @@
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the Great Expectations materializers."""
+
+import os
+from typing import Any, Dict, Type
+
+from great_expectations.checkpoint.types.checkpoint_result import (  # type: ignore[import]
+    CheckpointResult,
+)
+from great_expectations.core import ExpectationSuite  # type: ignore[import]
+from great_expectations.core.expectation_validation_result import (  # type: ignore[import]
+    ExpectationSuiteValidationResult,
+)
+from great_expectations.data_context.types.base import (  # type: ignore[import]
+    CheckpointConfig,
+)
+from great_expectations.data_context.types.resource_identifiers import (  # type: ignore[import]
+    ValidationResultIdentifier,
+)
+from great_expectations.types import SerializableDictDot  # type: ignore[import]
+
+from zenml.artifacts import DataAnalysisArtifact
+from zenml.materializers.base_materializer import BaseMaterializer
+from zenml.utils import yaml_utils
+from zenml.utils.source_utils import import_class_by_path
+
+ARTIFACT_FILENAME = "artifact.json"
+
+
+class GreatExpectationsMaterializer(BaseMaterializer):
+    """Materializer to read/write Great Expectation objects."""
+
+    ASSOCIATED_TYPES = (
+        ExpectationSuite,
+        CheckpointResult,
+    )
+    ASSOCIATED_ARTIFACT_TYPES = (DataAnalysisArtifact,)
+
+    @staticmethod
+    def preprocess_checkpoint_result_dict(
+        artifact_dict: Dict[str, Any]
+    ) -> None:
+        """Pre-processes a GE checkpoint dict before it is used to de-serialize a GE CheckpointResult object.
+
+        The GE CheckpointResult object is not fully de-serializable
+        due to some missing code in the GE codebase. We need to compensate
+        for this by manually converting some of the attributes to
+        their correct data types.
+
+        Args:
+            artifact_dict: A dict containing the GE checkpoint result.
+        """
+
+        def preprocess_run_result(key: str, value: Any) -> Any:
+            if key == "validation_result":
+                return ExpectationSuiteValidationResult(**value)
+            return value
+
+        artifact_dict["checkpoint_config"] = CheckpointConfig(
+            **artifact_dict["checkpoint_config"]
+        )
+        for result_ident, results in list(artifact_dict["run_results"].items()):
+            validation_ident = (
+                ValidationResultIdentifier.from_fixed_length_tuple(
+                    result_ident.split("::")[1].split("/")
+                )
+            )
+            validation_results = {
+                result_name: preprocess_run_result(result_name, result)
+                for result_name, result in results.items()
+            }
+            artifact_dict["run_results"][validation_ident] = validation_results
+            del artifact_dict["run_results"][result_ident]
```

Comment:
If I understand this code right, you're deleting all keys from `artifact_dict["run_results"]` and for each one add a new key back in. In that case, we could simply create another dict and then overwrite `artifact_dict["run_results"]`, like this:

```suggestion
        validation_dict = {}
        for result_ident, results in artifact_dict["run_results"].items():
            validation_ident = (
                ValidationResultIdentifier.from_fixed_length_tuple(
                    result_ident.split("::")[1].split("/")
                )
            )
            validation_results = {
                result_name: preprocess_run_result(result_name, result)
                for result_name, result in results.items()
            }
            validation_dict[validation_ident] = validation_results
        artifact_dict["run_results"] = validation_dict
```